File: NATSInternal.React/src/models/authentication/signInModel.ts
import { createCloneMethod } from "../baseModels";
import type { VerifyUserNameAndPasswordRequestDto } from "@/api";

declare global {
  type SignInModel = ClonableModel<{
    userName: string;
    password: string;
    toRequestDto(): VerifyUserNameAndPasswordRequestDto;
  }>;
}

export function createSignInModel(): SignInModel {
  const model: SignInModel = {
    $clone: createCloneMethod(() => model),
    userName: "",
    password: "",
    toRequestDto(): VerifyUserNameAndPasswordRequestDto {
      return {
        userName: this.userName,
        password: this.password
      };
    }
  };

  return model;
}

File: NATSInternal.Web/Models/Authentication/SignInModel.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using NATSInternal.Application.Localization;
using NATSInternal.Application.UseCases.Authentication;

namespace NATSInternal.Api.Models;

public class SignInModel
{
    #region Properties
    [DisplayName(DisplayNames.UserName)]
    public string UserName { get; set; } = string.Empty;
    
    [DisplayName(DisplayNames.Password)]
    [DataType(DataType.Password)]
    public string Password { get; set; } = string.Empty;
    #endregion

    #region Methods
    public VerifyUserNameAndPasswordRequestDto ToRequestDto()
    {
        return new()
        {
            UserName = UserName,
            Password = Password
        };
    }
    #endregion
}